### DISCUSSION TITLE
Unable to dynamically style a dataTable column based on data

### DISCUSSION THREAD

**Discussion by guedru** [primefaces/community#281] [PrimeVue]
Hello everyone,
I search every where but didn't find an answer to my problem.
I wish to set a background color to a cell depending on some value of the data.
We can access to row data in template #body but not in Column tag.
Here s what we can do : 
`<Column field="id">
       <template #body="{ data }">
                        {{ data.id }}
      </template>
</Column>`

And here is what I intend to do
`<Column field="id" :bodyStyle="getBackgroundColor(data)">...</Column>`

Is there a way to achieve this and access to row data within the Column tag ?

Thanks for your job.

**Comment by tugcekucukoglu** [comment]
Hello,

I can not replicate the problem. Could you share a reproducer link?

**Comment by furkanmturgut** [comment]
I couldn't find it either but


  ```
  <TColumn field="requestCount" sortable header="Çözülmemiş Talep Sayısı">
          <template #body="{ data }">
            <div :style="{ backgroundColor: data.requestCount === 0 ? 'springgreen' : data.requestCount > 3 ? 'red' : 'gold' }">
              <label style="display: flex; justify-content: center; font-weight: bold;">{{ data.requestCount }}</label>
            </div>
          </template>
        </TColumn>
```
Such an example may appear partially with use

**Reply by guedru** [reply]
Actually not, because you have padding inside the cell and just the background of the div is filled with color, not the entire cell. That can be done only if we can access to data in the parent node.

**Reply by guedru** [reply]
We can use this solution by removing the padding from the cell and add it in the div inside the template tag. But it would be great to have access to the data directly in the column tag :)

**Reply by foraqis** [reply]
Can you show an example of such code?
I recently encountered a similar problem and could not solve it.

**Comment by mustafadalga** [comment]
Hi @tugcekucukoglu ,
I have same problem. Will be a solution for future version ? I could not style entire column because Column has padding. As you see `.p-datatable .p-datatable-tbody>tr>td` has `padding:1rem`


```
      <Column
        v-for="col in selectedColumns"
        :key="col"
        :field="col"
        :header="col">

        <template #body="{ data }">
          <div :style="cellStyle(data, col)">
            {{ data[col] }}
          </div>
        </template>
      </Column>

```

<img width="667" alt="image" src="https://github.com/primefaces/community/assets/25087769/ecbfb78d-2001-41ce-a082-3e9359c75fda">

<img width="515" alt="image" src="https://github.com/primefaces/community/assets/25087769/a823d9ae-6144-4856-9dc5-77438bf6ca34">

**Reply by foraqis** [reply]
In my case this option worked
```
      <Column field=8 header="StTime"  style="padding: 0px; width: 7em;">
       <template #body="{ data }" >
          <div :style="cellClass(data[9])" style="padding: 8px;">
             {{data[8]}}
          </div>
        </template>
      </Column>
```

**Reply by slavco86** [reply]
you want to use `pt` and `ColumnPassThroughOptions`, specifically the `bodyCell` property.
This is based on the Lara preset I'm using in the unstyled mode:

```
bodyCell: ({ props, context, parent }) => ({
      class: [
        //Position
        { 'sticky box-border border-b': parent.instance.frozenRow },
        { 'sticky box-border border-b z-20': props.frozen },
        // Alignment
        'text-left',
        // Shape
        cellRowBorder,
        { 'first:border-l border-r border-b': context?.showGridlines },
        {
          'bg-surface-0 dark:bg-surface-800': parent.instance.frozenRow || props.frozen
        },
        // Spacing
        { 'p-2': context?.size === 'small' },
        { 'p-5': context?.size === 'large' },
        { 'p-4': context?.size !== 'large' && context?.size !== 'small' },
        // Color
        'border-off-black'
      ]
    }),
```

https://github.com/primefaces/primevue-tailwind/blob/main/presets/lara/datatable/index.js

**Reply by mustafadalga** [reply]
Thank you @slavco86 . I will briefly use solution of @guedru .
